projects/elements/src/lib/lazy-elements/lazy-elements-loader.service.ts

```diff
@@ -190,26 +178,18 @@ export class LazyElementsLoaderService {
     return url.replace(/https?:\/\//, '');
   }
 
-  private isPromise<T>(obj: T | Promise<T>): obj is Promise<T> {
-    return typeof (obj as any)?.then === 'function';
-  }
-
   private handleHook(hook: Hook, tag: string): Promise<void> {
     try {
-      const result = hook(tag);
-      if (this.isPromise(result)) {
-        return result;
-      } else {
-        return Promise.resolve();
-      }
+      return Promise.resolve(hook(tag));
```

Comment:
Promise.resolve creates a promise and resolve the value as a promise or not, so the `isPromise` is not needed as been it a promise or not it will return and be resolved in the then.